apps/website/package.json

```diff
@@ -0,0 +1,41 @@
+{
+  "name": "website",
+  "version": "0.1.0",
+  "private": true,
+  "scripts": {
+    "dev": "TAMAGUI_TARGET='web'  next dev",
+    "build": "TAMAGUI_TARGET='web'  next build",
```

Comment:
Fails on

```shell
felicio@felicios-mbp website % yarn dev
yarn run v1.22.19
$ TAMAGUI_TARGET='web'  next dev
ready - started server on 0.0.0.0:3000, url: http://localhost:3000
(node:8427) ExperimentalWarning: The Fetch API is an experimental feature. This feature could change at any time
(Use `node --trace-warnings ...` to show where the warning was created)

      Tamagui built config and components:

        Config     ./.tamagui/tamagui.config.cjs
        Components ./.tamagui/statusimcomponents-components.config.cjs
        
✘ [ERROR] Could not resolve "./names.native"

    ../../packages/components/dist/index.js:1:64985:
      1 │ ..._NAMES){const e=require("./names.native").names;Object.assign(Pt...
        ╵                            ~~~~~~~~~~~~~~~~

✘ [ERROR] Could not resolve "./names.native"

    ../../packages/components/dist/index.js:1:64985:
      1 │ ..._NAMES){const e=require("./names.native").names;Object.assign(Pt...
        ╵                            ~~~~~~~~~~~~~~~~

Tamagui Webpack Loader Error:
   Build failed with 1 error:
../../packages/components/dist/index.js:1:64985: ERROR: Could not resolve "./names.native"
Error: Build failed with 1 error:
../../packages/components/dist/index.js:1:64985: ERROR: Could not resolve "./names.native"
```

Reply:
@jkbktl got an idea why please?

Reply:
Not on a first look, I'll try dig deeper tomorrow.

Reply:
It seems like it should require this file into this file, but I don't understand why content of this file (`names.native`) is not already included in `dist/index.js` file. 🤔

Reply:
I believe we've briefly touched on a related topic during our "chill sesh",

- if you don't run dev/build for `packages/components`, remove or otherwise don't have the `dist` folder there
- and let nextjs transpile it with

```json
# packages/components/package.json
  "main": "./src/index.tsx",
  "types": "./src/index.tsx",
```

then it works.

But I still think that's only okay for local development or if didn't need to publish, as per

> You should never publish an internal package to npm.
>
> – https://turbo.build/blog/you-might-not-need-typescript-project-references#internal-typescript-packages

As to why, I suspect

- vite build step incorrectly eliminating https://github.com/tamagui/tamagui/blob/v1.11.1/packages/normalize-css-color/src/names.ts#L8-L11
- because even with `"build": "INCLUDE_CSS_COLOR_NAMES=0 TAMAGUI_TARGET='web' vite build",`
- it leaves this in the output

```js
// packages/components/dist/index.mjs
if (process.env.INCLUDE_CSS_COLOR_NAMES) {
  const e = require("./names.native").names;
  Object.assign(_t, e);
}
```

furthermore

- when later picked up by nextjs
- and if

```json
# packages/components/package.json
  "main": "./dist/index.js",
  "module": "./dist/index.mjs",
```

then `"dev": "INCLUDE_CSS_COLOR_NAMES=0 TAMAGUI_TARGET='web' next dev",` would still fail. This time perhaps due to nextjs's loader not respecting the `if (process.env.INCLUDE_CSS_COLOR_NAMES) {` condition and trying to call the `require` anyway. Something which seems parcel dealt with at https://github.com/parcel-bundler/parcel/pull/1166#issuecomment-386195313.

I have some ideas how to work around this, including

- patching
- pnpm
- reporting upstream

but cannot do it now.

Regardless, if you have any work based against this branch and given above, this shouldn't be blocking local development.

Reply:
Originally patched with https://github.com/status-im/status-web/pull/383/commits/3d668c22c1aa8d1e46cca21856976924558a1b5c and https://github.com/status-im/status-web/pull/383/commits/f2cb2a4dbba86f9ebc0c7ef45188d5c31e8e0302, later replaced by @prichodko's change at https://github.com/status-im/status-web/pull/383/commits/28efb7601f2f321c7f3537bf9630bafc24635750.